### PR TITLE
🐛 Destination BIgQuery can handle nulls inside arrays

### PR DESCRIPTION
**Pull request by edgao**
## What
As discovered in https://github.com/airbytehq/oncall/issues/295#issuecomment-1170036790, we previously would throw an error if we tried to extract an array which contained a null entry.

## How
Switch to the new-style JSON function, and handle NULLs. This necessitates also handling the new-style JSONPath syntax.

Example of old query: `select json_extract_string_array('{"foo": ["arst", "qwfp"]}', "$['foo']")`
Screen Shot 2022-07-07 at 7 48 55 PM

And example of new query:
```
SELECT ARRAY(
  SELECT IFNULL(x, "NULL")
  FROM UNNEST(json_value_array('{"foo": ["arst", "qwfp", null]}', '$."foo"')) AS x
)
```
Screen Shot 2022-07-08 at 11 26 01 AM

Explanation: `json_value_array` extracts the array into SQL values; `UNNEST` converts it into a table with a single column; we then replace nulls with a placeholder `"NULL"`, and convert it back into an array.

Note that the [test run](https://github.com/airbytehq/airbyte/pull/14522#issuecomment-1179262291) passed all normalization tests; `destination-postgres` tests are failing because the postgres container is refusing connections for some reason. I'm trying to figure out how to get tests passing, which will block the publish. (connectors build is also failing for some unrelated reason)

Also add entries to the expected_row_counts file, so that we actually verify the output exists. This test case was still useful previously (because it would have caught the dbt crash) but now we're properly checking that the rows were written. (we're still not actually verifying their contents though)

## Output Examples
This is the data that the _old_ macro produced: (I removed the `null` entries from `messages.txt` for this step, as it would have failed otherwise)
old_norm_top
old_norm_nested

And this is what the new data looks like. Note that it's basically identical, except that it handles `NULL` correctly.
new_norm_top
new_norm_nested

## Recommended reading order
1. json_operations.sql
2. catalog.json, messages.txt
3. everything else (i.e. the test output diff)

## 🚨 User Impact 🚨
I don't think so, but also couldn't figure out how to check the actual schema of the output.

### Community member or Airbyter

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [x] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [x] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/json_operations.sql [hunk 1]
```diff
@@ -249,5 +270,5 @@
 
 # https://cloud.google.com/bigquery/docs/reference/standard-sql/json_functions#json_extract_string_array
 {% macro bigquery__json_extract_string_array(json_column, json_path_list, normalized_json_path) -%}
-    json_extract_string_array({{ json_column }}, {{ format_json_path(normalized_json_path) }})
+    json_query_array({{ json_column }}, {{ bigquery_new_format_json_path(normalized_json_path) }})
```
Comment (grubberr) [inline airbyte-integrations/bases/base-normalization/dbt-project-template/macros/cross_db_utils/json_operations.sql:273]: I think you need to use equivalent counterpart:

OLD `JSON_EXTRACT_ARRAY` -> NEW `JSON_QUERY_ARRAY` - for array of objects
OLD `JSON_EXTRACT_STRING_ARRAY` -> NEW `JSON_VALUE_ARRAY` - for array of scalar values

**Comment by grubberr** [comment]
@edgao btw I have found in [docs](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#array_type) such option:

BigQuery raises an error if the query result has an ARRAY which contains NULL elements, although such an ARRAY can be used inside the query.

For example, this works:

```
SELECT FORMAT("%T", [1, NULL, 3]) as numbers;

+--------------+
| numbers      |
+--------------+
| [1, NULL, 3] |
+--------------+
```

I have tested it really works:

```
SELECT FORMAT("%T", json_value_array('{"a": ["a", "b", "c", null]}', "$.a"));


result:
["a", "b", "c", NULL]
```

**Comment by edgao** [comment]
I think that's returning a single string, which is the json representation of the array (i.e. `"[\"a\", \"b\", \"c\", NULL]"` rather than the array itself)?

**Review by grishick — APPROVED** [review]
LGTM!

**Pull request merged**